gapic-generator/templates/default/helpers/presenters/service_presenter.rb

```diff
@@ -176,10 +176,11 @@ def helpers_require
 
   def references
     @references ||= begin
-      m = @service.parent.messages.select { |m| m.fields.select(&:resource).any? }
-      pairs = m.map { |m1| [m1.name, m1.fields.map(&:resource).compact.first.pattern] }
-      pairs.sort_by! { |name, tmplt| name }
-      pairs.map { |name, tmplt| ResourcePresenter.new name, tmplt }
+      m = @service.parent.messages.select(&:resource)
+      m.sort_by! { |m1| m1.name }
+      # TODO: ResourceDescriptor#pattern is a list of patterns
```

Comment:
This is fine. Let's defer it. 

I think this [history bit](https://github.com/googleapis/api-common-protos/blob/2769b82d4993ccbe66a3d886aa5466fdbd050ea2/google/api/resource.proto#L104) is new too and I find it confusing.

Reply:
If/when this PS is merged I will open an issue to track this.